### PR TITLE
Handle compacting an empty delta

### PR DESCRIPTION
**Pull request by yankevn**
## Summary

This change handles the case where the delta results in no materialize results. In this case, committing the delta will be skipped in favor of directly writing back the RCF.

## Rationale

Empty deltas may occur, which will result in an assertion error during the merge deltas step while processing the merge results.

## Changes

List the major changes made in this pull request.

## Impact

Discuss any potential impacts the changes may have on existing functionalities.

## Testing

Describe how the changes have been tested, including both automated and manual testing strategies.
If this is a bugfix, explain how the fix has been tested to ensure the bug is resolved without introducing new issues.

## Regression Risk

If this is a bugfix, assess the risk of regression caused by this fix and steps taken to mitigate it.

## Checklist

- [ ] Unit tests covering the changes have been added
  - [ ] If this is a bugfix, regression tests have been added

- [x] E2E testing has been performed

## Additional Notes

Any additional information or context relevant to this PR.

### CODE REVIEW

File: deltacat/compute/compactor_v2/private/compaction_utils.py [hunk 1]
```diff
@@ -602,12 +605,26 @@ def _process_merge_results(
...
+            deltas,
+            stream_position=params.last_stream_position_to_compact,
+        )
+    else:
+        logger.info("No deltas found to merge, returning an empty delta...")
+        empty_manifest = Manifest.of(ManifestEntryList())
+        merged_delta: Delta = Delta.of(
+            DeltaLocator.of(
+                compacted_partition.locator,
+                None,
```
Comment (Zyiqin-Miranda) [inline deltacat/compute/compactor_v2/private/compaction_utils.py]: Is passing `None` as stream_position intended? Can you add test case assert follow-up incremental compaction can pick up this `None` stream position and conduct incremental starting from new delta correctly?

**Review by Zyiqin-Miranda — COMMENTED** [review]
On a high-level, any particular reason (I assume just for minimal changes) to commit an empty delta with empty manifests?

File: deltacat/compute/compactor_v2/private/compaction_utils.py (hunk 1 above)
Comment (yankevn) [reply deltacat/compute/compactor_v2/private/compaction_utils.py]: That was intended, but I'm realizing now that I got that from the sungate implementation which we shouldn't rely on in here, so I'll change it to use `current_time_ms()` instead.

**Comment by yankevn** [comment]
Yeah, committing an empty delta means that at least there's something committed vs an empty staged partition. When I initially tested by just skipping the committed delta, the follow-up job failed, which is why I added the empty delta.

**Review by Zyiqin-Miranda — APPROVED** [review]
Looks good, maybe good idea to add empty->incremental, incremental->empty->incremental test cases to guard the behavior, especially around stream_position/rcf fetching/deriving.
